src/contracts/Storage/CriteriaInterface.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\PersonalizationClient\Storage;
+
+interface CriteriaInterface
+{
+    /**
+     * @return array<string>
+     */
+    public function getItemTypeIdentifiers(): array;
+
+    /**
+     * @return array<string>
+     */
+    public function getLanguages(): array;
+
+    /**
+     * @return ?array<string>
+     */
+    public function getItemIds(): ?array;
```

Comment:
In what cases do we expect this method to return `null`?

If it's supposed to be a case for something being unsupported, then it should either be it's own interface or throw exceptions?

Reply:
My point is to add possibility to narrow down the results by given identifiers for specific item types and languages. I thought by default it should be null but if someone will want to fetch only specific items then they need to return array of item ids.

Reply:
Then it needs a comment in phpdoc explaining that returning `null` from this method has a specific meaning / significance.

Another option is creating a "hasser", which would enable us to drop the nullability of the return. In theory that's a "cleaner" solution, but I'm not sure if it's worth it?

Reply:
> My point is to add possibility to narrow down the results by given identifiers for specific item types and languages. I thought by default it should be null but if someone will want to fetch only specific items then they need to return array of item ids.

Could you provide an example of a code that would utilize this feature?

Reply:
After long thought I decided to drop this method. If someone will need this kind of feature then must implement it by it self.